File: source_code/project4_spring/main/src/main/java/com/service/main/entity/Transaction.java
package com.service.main.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Transaction {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private double amount;

    private String transactionType; // escrow, refund, webrevenue, hostrevenue

    private Date transferOn;

    @ManyToOne
    @JoinColumn(name = "bookingId")
    @JsonBackReference
    private Booking booking;

    @ManyToOne
    @JoinColumn(name = "userId")
    @JsonBackReference
    private User user;
}

File: source_code/project4_flutter/lib/features/property_detail/models/transaction.dart
import 'package:json_annotation/json_annotation.dart';
part 'transaction.g.dart';
@JsonSerializable()
class Transaction {

  final int? id;
  final int? bookingId;

  final double? amount;

  final String? transactionType; // escrow, refund, webrevenue, hostrevenue

  final DateTime? transferOn;
  Transaction({ this.bookingId, this.id, this.amount, this.transactionType, this.transferOn});
  factory Transaction.fromJson(Map<String, dynamic> json)=> _$TransactionFromJson(json);
  Map<String, dynamic> toJson() => _$TransactionToJson(this);

}